sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -1919,7 +1927,15 @@ public void testNullEmptyStringEquality() throws Exception
                   .dataSource(CalciteTests.DATASOURCE1)
                   .intervals(querySegmentSpec(Filtration.eternity()))
                   .granularity(Granularities.ALL)
-                  .filters(expressionFilter("case_searched((\"dim2\" == 'a'),1,isnull(\"dim2\"))"))
+                  .filters(
+                      or(
+                          selector("dim2", "a", null),
+                          and(
+                              selector("dim2", null, null),
+                              not(selector("dim2", "a", null))
+                          )
+                      )
+                  )
```

Comment:
The new filter is `dim2 = 'a' OR (dim2 IS NULL AND dim2 != 'a')`. It seems a bit… weird. Ideally it should be simplified to `dim2 = 'a' OR dim2 IS NULL`. It's fine for now, but could you add a comment about this?

Reply:
Added a comment on the unnecessary `dim2 != 'a'`